### DISCUSSION TITLE
Problems getting good speech to text for auctions and fundraisers

### DISCUSSION THREAD

**Discussion by johnnydanger** [deepgram/community#957] [General help] [answered]
The issue is that Deepgram consistently incorrectly transcribes an auctioneer calling dollar amounts.  

For example, the audio "seven thousand, seven, seventy five hundred, seven five, seventy five hundred, seven thousand, do I have seventy five hundred?  At seven thousand, do I have seventy five hundred?" transcribes as follows:

7 1000, 700,75100, 75, 7500, 10000, 7000.
7 1000.
Do I have 75100? At 7000, do I have 75100?

I find it particularly odd that Deepgram will transcribe "seventy five hundred" as "75100".  It does this often!  no space, no comma, not properly transcribed without say "thousand" after "seven".  Any ideas?  The current settings are:

const deepgram = deepgramClient.listen.live({
      model: "enhanced",
      language: "en-US",
      smart_format: true,
      interim_results: true
   });

**Comment by jpvajda** [comment]
@johnnydanger  could you try this with our[ Nova-2](https://developers.deepgram.com/docs/model#custom) model? it might handle it better.  Though numbers can always be tricky for STT, and the speed at which the auctioneer is saying the numbers could also make this somewhat challenging to get right. 

Depending on if there are breaks between  you might want to look at[ endpointing](https://developers.deepgram.com/docs/endpointing) with interim results. Also check out our docs on [utterance_end](https://developers.deepgram.com/docs/utterance-end) and [speech_started](https://developers.deepgram.com/docs/speech-started)

Playing with these different features of our Streaming STT API might get you what you need, but it might take some experimentation.

**Reply by johnnydanger** [reply]
Thanks for the tips.  I found that the enhanced model seemed to do better otherwise, but didn't notice the number issue prior to changing to Enhanced so I'll play with Nova-2

Granted the use case/stream of audio is atypical in a general transcription sense, but is of course very common in the auction world.  Regardless, I find it quite odd that saying "sixty five hundred" always transcribes to 65100.  Always.  At least with the 'enhanced' model.  It seems complete normal to say "sixty five hundred" in normal speech—like if you were talking about a bank account balance for instance.

In any case, I just tested Deepgram using Nova-2 and said clearly and slowly, "seventy five hundred.  sixty five hundred. six thousand. sixty five hundred. seven thousand.  seventy five hundred." and it transcribed the audio as follows:

75100, 65100, 6000, 65100, 7,000, 75100.

Unfortunately Nova-2 doesn't solve the issue.  Any other thoughts on a solution?

**Reply by jpvajda** [reply]
@johnnydanger  you might get some value out of using[ keywords](https://developers.deepgram.com/docs/keywords) you could try to pass in numbers as keywords and try boost those numerical values you are trying to get the model to handle better.

Setting the keyword intensifier higher might help, so just play around with that 

```curl
curl \
  --request POST \
  --header 'Authorization: Token YOUR_DEEPGRAM_API_KEY' \
  --header 'Content-Type: audio/wav' \
  --data-binary @youraudio.wav \
  --url 'https://api.deepgram.com/v1/listen?keywords=6500:1'

**Reply by johnnydanger** [reply]
Interesting suggestion.  I gave it a try and it does indeed solve things... kind of.  I put in the keywords: "4500", "5500", "6500", "7500", and now those transcribe properly.  EXCEPT!  Now when I say "eighty five hundred" is transcribes as "4500" every single time.  Bah.  Then I experimented more and found that by adding numbers as keywords gives a lot of random unexpected results.  Now when I say "fifteen thousand" for instance it returns "6500", also every time.  So alas, using numbers as keywords isn't a functional solution.

I wonder if there's any possibility of Deepgram altering how numbers are transcribed.  It seems like a serious weak point, particularly if it was to be used as an automated solution for banks, auctions, or anything relating to saying a number.

**Reply by johnnydanger** [reply]
Just did a bit more experimenting, and found that "smart_format" is the issue when it comes to numbers.  The transcription is great with smart format turned off—however doing so loses all the punctuation, line breaks, etc. so I would have to add more processing in my app to get that stuff back.  Not the end of the world, though unfortunate.  I wonder if there are any secret parameters that turn off number formatting when using smart_format?  Because that would solve things tout suite!

**Reply by jpvajda** [reply, marked as answer]
@johnnydanger  there are actually. try using `numerals` or `punctuate` and set to `true`

https://developers.deepgram.com/docs/numerals
https://developers.deepgram.com/docs/punctuation

These are the older formatting features that `smart_format` attempted to replace, but I know sometimes they work better independently

**Comment by johnnydanger** [comment]
Awesome, thanks John! I’ll dig into it and let you know the solution I land on. Thanks for all the help.
